File: Retrofit2/app/src/main/java/com/example/retrofit2/MainActivity.java
package com.example.retrofit2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText etName;
    EditText etEmail;
    Button btnCreate;
    Button btnSearch;
    Button btnDelete;
    MainActivityViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViewModel();
        btnCreate = findViewById(R.id.createButton);
        btnSearch = findViewById(R.id.searchButton);
        btnDelete = findViewById(R.id.deleteButton);
        etName = findViewById(R.id.etName);
        etEmail = findViewById(R.id.etEmail);

        btnCreate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                createNewUser();
            }
        });

        btnSearch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getUser();
            }
        });

        btnDelete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                deleteUser();
            }
        });
    }

    private void createNewUser() {
        String name = etName.getText().toString();
        String email = etEmail.getText().toString();
        User user = new User("", name, email, "Active", "Male");
        viewModel.createNewUser(user);
    }

    private void getUser() {
        String id = etName.getText().toString();
        viewModel.getUser(id);
    }

    private void deleteUser() {
        String id = etName.getText().toString();
        viewModel.deleteUser(id);
    }


    private void initViewModel() {
        viewModel = new ViewModelProvider(this).get(MainActivityViewModel.class);
        viewModel.getCreateUserObserver().observe(this, new Observer<UserResponse>() {
            @Override
            public void onChanged(UserResponse userResponse) {
                if (userResponse == null) {
                    Toast.makeText(MainActivity.this, "failed to create new user", Toast.LENGTH_LONG).show();
                } else {
                    Toast.makeText(MainActivity.this, "Successfully created new user", Toast.LENGTH_LONG).show();
                }
            }
        });
    }

}

File: ImageViewKt/app/src/main/java/com/example/imageviewkt/MainActivity.kt
package com.example.imageviewkt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.example.imageviewkt.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private var mBinding : ActivityMainBinding?=null
    private val binding get() = mBinding!!

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)
        mBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        
        binding.btnToast.setOnClickListener {
            //ImageView 이미지를 setImageResource라는 함수를 이용하여 이미지를 변경이 가능하다
            binding.ivProfile.setImageResource(R.drawable.androidlogo)
            //Toast라는 것을 이용하여 알림이 가능하게 해준다
            //첫 번째 인자 : 화면, 두번째 인자 : 메세지, 세 번째 인자 : Toast 노출 길이
            //show라는 함수를 꼭 붙여 주어야한다.
            Toast.makeText(this@MainActivity, "버튼이 클릭 되었습니다.", Toast.LENGTH_SHORT).show()
        }
    }
}